src/lib/fy-utils.c

```diff
@@ -15,16 +15,101 @@
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#if defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
 #include <termios.h>
 #include <unistd.h>
 #include <sys/select.h>
 #include <sys/time.h>
 #include <sys/types.h>
+#elif defined(_MSC_VER)
+#include <windows.h>
+#endif
 
 #include "fy-utf8.h"
 #include "fy-ctype.h"
 #include "fy-utils.h"
 
+int fy_get_pagesize() {
+#if defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
+       return sysconf(_SC_PAGESIZE);
+#elif defined (_MSC_VER)
+    SYSTEM_INFO si;
+    GetSystemInfo(&si);
+       return si.dwPageSize;
+#endif
+}
+
+#if defined(_MSC_VER)
+#ifndef VA_COPY
+# ifdef HAVE_VA_COPY
+#  define VA_COPY(dest, src) va_copy(dest, src)
+# else
+#  ifdef HAVE___VA_COPY
+#   define VA_COPY(dest, src) __va_copy(dest, src)
+#  else
+#   define VA_COPY(dest, src) (dest) = (src)
+#  endif
+# endif
+#endif
+
+#define INIT_SZ 128
+
+int
+vasprintf(char **str, const char *fmt, va_list ap)
```

Comment:
Maybe you're over doing it here.

IIRC vsnprintf with a NULL string returns the exact size, so you can just call it twice, first call to find the size, second call after allocating it.